### ISSUE TITLE
macOS: do not prevent logoff

### ISSUE THREAD

**Issue by jewesta** [microsoft/vscode#173272]
Type: <b>Bug</b>

This problem is bugging me for some time now as I need to switch between user accounts on my Mac regularly. Today I tried to reproduce it and I think I found a way to do that.

* Logon to the Mac
* Start VSCode
* Open any file (this is important!)
* Close all VSCode tabs and the main application window. VSCode still needs to be running in the background (dot below dock icon)
* Attempt to log off

Mac OS will now do nothing. Manually closing VSCode via the dock icon will immediately resolve the issue and Mac OS will log the user off.

I am on Monterey 12.6.2

VS Code version: Code 1.75.0 (e2816fe719a4026ffa1ee0189dc89bdfdbafb164, 2023-02-01T15:24:42.903Z)
OS version: Darwin x64 21.6.0
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz (16 x 2400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|14, 9, 6|
|Memory (System)|16.00GB (3.93GB free)|
|Process Argv|--crash-reporter-id 241582ba-c5d4-49e6-b5b9-563fb02348a7|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (4)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-diff|fab|1.4.2
vscode-language-pack-de|MS-|1.75.2023020109
hexeditor|ms-|1.9.11
openhab|ope|1.0.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593:30376534
pythonvs932:30410667
cppdebug:30492333
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
azure-dev_surveyone:30548225
vscccc:30610679
pyindex848cf:30577861
nodejswelcome1cf:30587006
282f8724:30602487
pyind779cf:30657577
f6dab269:30613381
pythonsymbol12cf:30657549
2i9eh265:30646982

```

</details>

**Comment by VSCodeTriageBot** [comment]
This feature request is now a candidate for our backlog. The community has 60 days to [upvote](https://github.com/microsoft/vscode/wiki/Issues-Triaging#up-voting-a-feature-request) the issue. If it receives 20 upvotes we will move it to our backlog. If not, we will close it. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by jewesta** [comment]
Some additional Info:

I have since updated to macOS Ventura 13.2 and the problem persists. I can reproduce it with the steps I described in the OP. After attempting to logoff (where nothing happens) there is a timeout where the OS waits for the applications to self terminate.

After the timeout (60s?) macOS presents this message.
 
<img width="480" alt="Bildschirm­foto 2023-02-07 um 20 39 42" src="https://user-images.githubusercontent.com/25198212/217348445-d2f279d9-a1e0-40ed-b7b8-95323b99b7f2.png">

It is in German, so I'll put the english equivalent here:

Your Mac hasn't logged out because "Code" failed to quit. To continue logging out, quit "Code".
To quit an unresponsive app, select Force Quit from the Apple Menu (or press Command-Option-Esc).
